mods/Windows/OnlineCTR/Network_PC/Server/SV_main.c

```diff
@@ -702,6 +719,108 @@ void ServerState_Tick()
 	}
 }
 
+void InitializeBadWordList(BadWordList* list) {
+	list->word_count = 0;
+
+	AddBadWord(list, "Ass");
+	AddBadWord(list, "Asshole");
+	AddBadWord(list, "Nigga");
+	AddBadWord(list, "Nigger");
+	AddBadWord(list, "Whore");
+	AddBadWord(list, "Fuck");
+	AddBadWord(list, "Bitch");
+	AddBadWord(list, "Bastard");
+	AddBadWord(list, "Pussy");
+	AddBadWord(list, "Tits");
+	AddBadWord(list, "Titty");
+	AddBadWord(list, "Fucker");
+	AddBadWord(list, "Porn");
+	AddBadWord(list, "Penis");
+	AddBadWord(list, "Vulva");
+	AddBadWord(list, "Anus");
+	AddBadWord(list, "Anal");
+	AddBadWord(list, "Sex");
+	AddBadWord(list, "Balls");
+	AddBadWord(list, "Piss");
+	AddBadWord(list, "Orgasm");
+	AddBadWord(list, "Wank");
+	AddBadWord(list, "Masturbate");
+	AddBadWord(list, "P0rn");
+	AddBadWord(list, "Twat");
+	AddBadWord(list, "Spic");
+	AddBadWord(list, "Shemale");
+	AddBadWord(list, "Cum");
+	AddBadWord(list, "Fag");
+	AddBadWord(list, "Faggot");
+	AddBadWord(list, "Dick");
+	AddBadWord(list, "Ejaculat");
+	AddBadWord(list, "Cunt");
+	AddBadWord(list, "Coom");
+	AddBadWord(list, "Clit");
+	AddBadWord(list, "Cock");
+	AddBadWord(list, "C0ck");
+	AddBadWord(list, "Bum");
+	AddBadWord(list, "Boob");
+	AddBadWord(list, "Boner");
+	AddBadWord(list, "Pussies");
+	AddBadWord(list, "Damn");
+	AddBadWord(list, "Cunnilingus");
+	AddBadWord(list, "Deepthroat");
+	AddBadWord(list, "Arse");
+	AddBadWord(list, "Puke");
+	AddBadWord(list, "Puta");
+	AddBadWord(list, "Smegma");
+	AddBadWord(list, "Prick");
+	AddBadWord(list, "Smut");
+	AddBadWord(list, "Pube");
+	AddBadWord(list, "Fart");
+	AddBadWord(list, "Gay");
+	AddBadWord(list, "Lesbian");
+	AddBadWord(list, "Rectum");
+	AddBadWord(list, "Retard");
+	AddBadWord(list, "Semen");
+	AddBadWord(list, "Shart");
+	AddBadWord(list, "Shit");
+	AddBadWord(list, "Teet");
+	AddBadWord(list, "Twink");
+	AddBadWord(list, "Vagina");
+}
+
+void AddBadWord(BadWordList* list, const char* word) {
+	if (list->word_count < BADWORDS_COUNT) {
+		list->words[list->word_count] = _strdup(word);
+		list->word_count++;
+	}
+	else {
+		printf("Maximum badword list length of %d reached", BADWORDS_COUNT);
+	}
+}
+
+bool WordContainsBadWord(const BadWordList* list, const char* word) {
+	char lowerCaseWord[USERNAME_MAX_LENGTH];
+	strncpy(lowerCaseWord, word, USERNAME_MAX_LENGTH);
+	StringToLowerCase(lowerCaseWord);
+
+	for (int i = 0; i < list->word_count; i++) {
+		char lowerCaseBadWord[BADWORD_MAX_LENGTH];
+		strncpy(lowerCaseBadWord, list->words[i], BADWORD_MAX_LENGTH);
+		StringToLowerCase(lowerCaseBadWord);
```

Comment:
if all bad words will eventually be converted to lower case, maybe it's better to already define them in lower case and skip this step here